packages/components/src/custom-select-control/index.js

```diff
@@ -69,14 +76,12 @@ export default function CustomSelectControl( {
 		highlightedIndex,
 		selectedItem,
 	} = useSelect( {
-		initialSelectedItem: items[ 0 ],
+		initialSelectedItem: items[ valueIndex >= 0 ? valueIndex : 0 ],
 		items,
 		itemToString,
 		onSelectedItemChange,
```

Comment:
```suggestion
		onSelectedItemChange,
		selectedItem: valueIndex >= 0 ? items[ valueIndex ] : undefined
```

Downshift needs to be aware when the item really does change.

Reply:
I'm afraid that's the one thing we can't do, because it causes Downshift to announce changes to the selection whenever the component re-renders. Apart from the minor issue of the highlight not always matching the selected item (which we may be able to fix by leveraging `:focus` styles), is there anything else that won't work with these changes? Otherwise I think we'll have to use a different library, or implement our own solution.

Reply:
If the component rerenders and `value` is referentially equal, then it shouldn't be announcing anything, right? I think the issue is that the value is being recreated on render.

Have you tried memoizing `getSelectOptions` in the font size picker?

Reply:
Hmm, I haven't. But that still won't solve the problem of changes being announced whenever the value changes outside of the component, e.g. if I select a random value in the font size number input, it announces "custom has been selected".

Reply:
>I select a random value in the font size number input, it announces "custom has been selected".

It would only happen once when the value is switched to custom. Is that still worth avoiding?

Not passing Downshift the value means that when it changes elsewhere, Downshift won't know about it. If I understand how it works correctly, this means that all the prop getters will return props based on an outdated selected item. E.g., entering a custom value would change the rendered label because you are getting the value manually for that. Still, all the prop getters won't know about it internally so that they can return the relevant props.

Reply:
> If I understand how it works correctly, this means that all the prop getters will return props based on an outdated selected item.

Yeah, that's why the highlighting isn't working correctly when the value is updated elsewhere. With all the other props, I've managed to work around. 

I'll try memoizing and if it solves the main issue, it's probably good enough for now. But we should revisit this later, as having that unexpected announcement when using the number input isn't really great.

Reply:
Maybe we can add a prop to "silence" the announcement in those cases?

Reply:
Do you have anything specific in mind? There doesn't seem to be a way of silencing the announcement in Downshift, and I can only think of extremely dodgy solutions to do it on our side 😅 (like removing Downshift's `aria-live` container or trying to override it with an empty `speak` - not sure that'd even work btw)

Reply:
Update: memoizing `getSelectOptions` works for the announcement when switching editor modes, so that's the main issue solved ✅

Reply:
>(like removing Downshift's aria-live container or trying to override it with an empty speak - not sure that'd even work btw)

That was my line of thinking as well, haha.

>works for the announcement when switching editor modes, so that's the main issue solved

Awesome!